websites/S/Sonic Retro/presence.ts

```diff
@@ -0,0 +1,285 @@
+					presenceData.state = `${document.title.split(" - Sonic Retro")[0]}`;
+					presenceData.buttons = [
+						{
+							label: "View Special Page",
+							url: location.href
+						}
+					];
+				}
+			} else if (document.title.startsWith("Category:")) {
+				presenceData.largeImageKey = "reading";
+				presenceData.details = "Reading an article:";
+				presenceData.state = document.title
+					.split("Category:")[1]
+					.split(" - Sonic Retro")[0];
+				presenceData.buttons = [
+					{
+						label: "View Article",
+						url: location.href
+					}
+				];
+			} else if (document.title.startsWith("Talk:")) {
+				presenceData.largeImageKey = "reading";
+				presenceData.details = "Browsing a talk page:";
+				presenceData.state = `${
+					document.title.split("Talk:")[1].split(" - Sonic Retro")[0]
+				}`;
+			} else if (document.title.startsWith("Sonic Retro:")) {
+				presenceData.largeImageKey = "reading";
+				presenceData.details = "Reading an article:";
+				presenceData.state = `${
+					document.title.split("Sonic Retro:")[1].split(" - Sonic Retro")[0]
+				}`;
+				presenceData.buttons = [
+					{
+						label: "View Article",
+						url: location.href
+					}
+				];
+			} else if (document.title.startsWith("User:")) {
+				presenceData.largeImageKey = "search";
+				presenceData.details = "Viewing a user's page:";
+				presenceData.state = `@${
+					document.title.split("User:")[1].split(" - Sonic Retro")[0]
+				}`;
+				presenceData.buttons = [
+					{
+						label: "View User",
+						url: location.href
+					}
+				];
+			} else if (
+				pathname.includes("/File:" || "/Sonic_Retro:General_disclaimer")
+			) {
+				presenceData.largeImageKey = "logo_1024";
+				presenceData.details = "Viewing an unsupported page";
+			} else if (pathname === "/index.php" && search.includes("search=")) {
+				presenceData.largeImageKey = "search";
+				if (document.title.startsWith("Search -"))
+					presenceData.details = "Searching for something...";
+				else {
+					presenceData.details = `Searching for ${
+						document.title
+							.split("Search results for ")[1]
+							.split(" - Sonic Retro")[0]
+					}`;
+				}
+			} else if (
+				pathname === "/index.php" ||
+				pathname.startsWith("User_talk:")
+			) {
+				presenceData.largeImageKey = "logo_1024";
+				presenceData.details = "Viewing an unsupported page";
+			} else {
+				presenceData.largeImageKey = "reading";
+				presenceData.details = "Reading an article:";
+				presenceData.state = `${document.title.split(" - Sonic Retro")[0]}`;
+				presenceData.buttons = [
+					{
+						label: "Read Article",
+						url: location.href
+					}
+				];
+			}
+			break;
+		}
+		case "forums.sonicretro.org": {
+			presenceData.smallImageKey = "forums_512";
+			if (search) {
+				const forumTitle = document.title.split(
+					" | Sonic and Sega Retro Forums"
+				)[0];
+				if (search.includes("?watched/forums")) {
+					presenceData.largeImageKey = "search";
```

Comment:
This is repeated 13 time along this presence, surely you can spare some of those

Reply:
Figuring an approach to this review; please bear with me.

Reply:
just set it outside the if and inside this case